### PR TITLE
Fix bugs found by SonarLint

### PR DESCRIPTION
**Pull request by angyaloliver**
Fixed 3 bugs found by SonarLint.

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -14,34 +14,35 @@
 
   private int torpedoCount = 0;
 
-  public TorpedoStore(int numberOfTorpedos){
+  private Random generator = new Random();
```
Comment (schmelczerandras) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:17]: Couldn't we place this definition inside the class?

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -14,34 +14,35 @@
...
         FAILURE_RATE = Double.parseDouble(failureEnv);
-      } catch (NumberFormatException nfe) {
+      } catch (final NumberFormatException nfe) {
         FAILURE_RATE = 0.0;
       }
     }
   }
 
-  public boolean fire(int numberOfTorpedos){
+  public boolean fire(final int numberOfTorpedos){
```
Comment (schmelczerandras) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:33]: I really like your use of finals!

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -14,34 +14,35 @@
...
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (schmelczerandras) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:45]: Oh, nice catch!

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 2 above)
Comment (angyaloliver) [reply src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:33]: Oops.. 👀

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 1 above)
Comment (angyaloliver) [reply src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:17]: Thanks, I'll look into it! 🤔🤔

**Pull request merged**